### ISSUE TITLE
Add-in won't run due to certificates error

### ISSUE THREAD

**Issue by kbrandl** [OfficeDev/generator-office#486]
I just used Yo Office to create an Excel add-in. When I run `npm start` the server starts successfully (no error messages), the message that's written to the console says `You already have trusted access to https://localhost.`, and the `C:\Users\kibrandl\.office-addin-dev-certs\localhost.crt` file that's mentioned in the console message does indeed exist (although with a date stamp from a few weeks ago).

![image](https://user-images.githubusercontent.com/3019529/64272446-3e0ea500-cef4-11e9-93d6-2babb4757017.png)

However, when I attempt to open the add-in in Excel, it won't load successfully and shows the following certificate-related error message:

![image](https://user-images.githubusercontent.com/3019529/64272535-6ac2bc80-cef4-11e9-940d-f0af317f6197.png)

Likewise, the add-in won't run in Excel Online (in Chrome):

![image](https://user-images.githubusercontent.com/3019529/64272793-e7ee3180-cef4-11e9-9220-5eaa026ee33e.png)

If I load `https://localhost:3000` (or `https://localhost:3000/taskpane.html`) in a browser (Chrome), I also see a cert-related error message:

![image](https://user-images.githubusercontent.com/3019529/64272622-9ba2f180-cef4-11e9-88b9-a156a48fa2cb.png)

Info about my environment:
- Issue observed in both:
    - Excel on Windows Version 1908 (build 11929.20254)
    - Excel Online (using Chrome browser)
- node v 10.15.3
- npm v 6.4.1
- generator-office v 1.3.0
- yo v 3.1.0

Are you able to repro this issue, and any suggestions on how I might resolve it?

**Comment by akrantz** [comment]
Run: `npx office-addin-dev-certs uninstall` and then `npx office-addin-dev-certs install`. This will recreate the certificates.
If you're still having problems, try restarting your computer.

**Comment by kbrandl** [comment]
@akrantz thanks for the feedback; unfortunately it didn't resolve the cert-related error I'm seeing. Here's what I did:
1. Ran both commands (uninstall, then install certs).
2. Ran npm start and tried to load my add-in in Excel -> same cert-related error occurs.
3. Restarted my computer.
4. Ran npm start and tried to load my add-in in Excel -> same cert-related error occurs.
5. Ran both commands (uninstall, then install certs). 
6. Ran npm start and tried to load my add-in in Excel -> same cert-related error occurs.

Any other suggestions? FWIW, @AlexJerabek was unable to repro this issue on his machine -- so it seems like it's a problem with my local environment.

**Comment by akrantz** [comment]
I think the error might be the wrong error message for the problem. Perhaps its the Edge webview localhost access. Try running `npx office-addin-dev-settings appcontainer EdgeWebView --loopback`

**Comment by kbrandl** [comment]
Thanks -- I actually already tried that, didn't make a difference. I'm seeing this same error in Excel Online using Chrome -- so it doesn't seem specific to the Edge WebView control.

**Comment by akrantz** [comment]
With the dev server is running, does https://localhost:3000/taskpane.html work in the browser?

**Comment by kbrandl** [comment]
No -- I also get the cert-related error in the browser (Chrome). (See screenshot I uploaded earlier in this thread.)

**Comment by Rothammel** [comment]
i have the same problem. 
i follow this tutorial: https://docs.microsoft.com/de-de/office/dev/add-ins/tutorials/word-tutorial

the certificate in Start/certs folder is from localhost-ca but internet explorer shows on https://localhost:3000/ a cert from  Internet Widgits Pty Ltd

**Comment by TCourtneyOwen** [comment]
I will take a look at this today to see if I can get a repro.

**Comment by TCourtneyOwen** [comment]
I **can't** get a repro of this after completely uninstalling certs and then running "npm run start", which prompts me to install the certs again as part of starting the dev-server.  One thing I would check is to see if there is a .office-addin-dev-certs folder getting created in your user profile folder (e.g. C:\Users\cowen\.office-addin-dev-certs) and that it contains the certs.

**Comment by akrantz** [comment]
I doubt that this is an issue with office-addin-dev-certs. I suspect that there are other certs related to https://localhost on the machine and these are not properly signed/trusted. 

@Rothammel If IE is using a different cert which isn't properly trusted or is self-signed, perhaps you should remove it.

@kbrandl Could you check the certificate details as well?

**Comment by kbrandl** [comment]
@akrantz yes -- I was just examining the certs on my machine when I noticed this message from you. And, in fact, there were two "localhost" certs on my machine:

![image](https://user-images.githubusercontent.com/3019529/64377658-bf8f3180-cfdf-11e9-8f1a-bbe528ff92d2.png)

After I removed these certs, everything worked fine (the cert-related error went away). 

@Rothammel I'd suggest you examine the **Trusted Root Certification Authorities** certificates on your machine -- sort the list by the **Issued To** column -- if you see any certificates issued to **localhost** or **localhost-ca** delete them. Then, try running `npm start` again to start localhost and (hopefully) the certificate-related error will no longer occur.

**Comment by Rick-Kirkham** [comment]
@Rothammel This fixed it on my machine. Could you see if it fixes it for you? 
1. Close all browsers and all Office apps.
2. If the server is running, shut it down by entering Ctrl-C twice. Then close the command prompt window.
3. Open the bconfig.json file in the root of the project.
4. Replace the line `"https": true,` with the following:

```
"https": {
        "key": "./certs/server.key",
        "cert": "./certs/server.crt"
    },
```

5. Repeat the "Test the add-in" section of the tutorial.

**Comment by tinker-devdav** [comment]
This tool does not work for me on Windows 11.  I managed to get it working by following some of the examples here to get the certificate installed https://learn.microsoft.com/en-us/dotnet/core/additional-tools/self-signed-certificates-guide 

When I run the command "npx office-addin-dev-certs install" it came back with success. but the certificate was never installed in the registry.  I then manually load the certificate and it still doesn't work.

**Comment by alexgschwend** [comment]
I faced a similar problem and after trying numerous solutions, what eventually resolved it for me was updating to version 1.11.4 of office-addin-dev-certs.
